### PR TITLE
Adds seeking functionality

### PR DESCRIPTION
**Pull request by emilymdubois**
This PR tracks the ability to seek to start emitting events from a point in the route determined by the user. This functionality is desirable for use cases such as [`directions-sim`](https://github.com/mapbox/directions-sim), where you may want to be able to rewind or fast-forward through a route.

Other functionalities I'd like to add before merging:
- [x] Current step locator
  - This will be helpful in interacting with `navigation.js`, which requires a current step input to display (1) upcoming maneuvers, and (2) distance from said maneuver

### CODE REVIEW

**Comment by emilymdubois** [comment]
:wave: @emilymcafee mind taking a quick 👀  through to see if anything looks potentially disruptive?

File: lib/event-emit.js [hunk 1]
```diff
@@ -22,19 +24,28 @@ function Emitter(geojson, updateInterval) {
...
+
+  this.setTime = function(start) {
+    stepsTaken = start / updateInterval;
+    return stepsTaken;
+  };
 
   /**
    * Return the next location event for a route.
    *
+   * @param {Number} optional stepsTaken value for seeking purposes.
```
Comment (emilymcafee) [inline lib/event-emit.js]: is stepsTaken intended to be a parameter of the `next()` function? Or are you using it globally?

e.g. `this.next = function(stepsTaken)` vs `this.next = function()`

I think including it in documentation here ^ implies the first.

File: lib/event-emit.js [hunk 2]
```diff
@@ -12,6 +12,8 @@ Emitter.EmitterPlace = EmitterPlace;
  * @param {Object} geojson Timestamped geojson of route.
  * @param {number} updateInterval Interval (milliseconds) at which
  * to generate location events.
+ * @param {Number} start Optional index for geojson.geometry.coordinates at
```
Comment (emilymcafee) [inline lib/event-emit.js]: move this ^ down to comment above `this.setTime = function(start)` ? (instead of Emitter)

**Comment by emilymcafee** [comment]
👋 I'm wondering if we should use a more descriptive property name than `.time`, since this is tracking # of steps taken (more like an index) and not the amount of time that has passed. That tripped me up initially - am I reading it properly?

File: lib/event-emit.js (hunk 1 above)
Comment (emilymdubois) [inline lib/event-emit.js]: Whoops, that was a vestige from an experiment, I'm 🙅  it

File: lib/event-emit.js (hunk 2 above)
Comment (emilymdubois) [inline lib/event-emit.js]: 👍

**Comment by emilymdubois** [comment]
:wave: @emilymcafee, you're right, it is tracking the number of steps. I initially called it `setTime` because inputs a start time and outputs number of steps taken, but I think it could benefit from some clarity. Just pushed some semantic fixes, let me know if they make sense 🙇

**Comment by emilymcafee** [comment]
Awesome, looks great! 👍

**Comment by emilymdubois** [comment]
bitmoji

**Pull request merged**